03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,23 +1,52 @@
 const Task = require('../models/task');
+const asyncWrapper = require('../middleware/async');
+const { createCustomError } = require('../errors/custom-error');
+const getAllTasks = asyncWrapper(async (req, res) => {
+  const tasks = await Task.find({});
+  res.status(200).json({ tasks: tasks });
+  // res.status(200).json({tasks, amount: tasks.length})
+  //  res .status(200).json({ success: true, data: { tasks, nbHits: tasks.length } });
+});
 
-const getAllTasks = (req, res) => {
-  res.send('all tasks');
-};
-
-const createTask = async (req, res) => {
+const createTask = asyncWrapper(async (req, res) => {
   const task = await Task.create(req.body);
   res.status(200).json({ task });
-};
-
-const getTask = (req, res) => {
-  res.json({ id: req.params.id });
-};
-const updateTask = (req, res) => {
-  res.send('update task');
-};
-
-const deleteTask = (req, res) => {
-  res.send('delete task');
-};
+});
+
+const getTask = asyncWrapper(async (req, res, next) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOne({ _id: taskID });
+
+  if (!task) {
+    return next(createCustomError(`No task with ID ${taskID}`, 404));
+    // return res.status(404).json({ msg: `No task with ID ${taskID}` });
+  }
+  res.status(200).json({ task });
+});
+
+const updateTask = asyncWrapper(async (req, res) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
+    new: true,
+    runValidators: true,
+  });
+  if (!task) {
+    return next(createCustomError(`No task with ID ${taskID}`, 404));
+  }
+
+  res.status(200).json({ task });
+});
+
+const deleteTask = asyncWrapper(async (req, res) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOneAndDelete({ _id: taskID });
+
+  if (!task) {
+    return next(createCustomError(`No task with ID ${taskID}`, 404));
+  }
+  // res.status(200).json({ task });
+  // res.status(200).send()
+  res.status(200).json({ task: null, status: 'success' });
```

Comment:
I think this should return the actual task instead of null here. Also, you don't need the `status: 'success'` bit.